### PR TITLE
connection status service signal implementation

### LINKED ISSUE #17008: Angular
- [ ] Create a service store internet connection status.(use InternalStore<> )
- [ ] create a component and add app.component.ts (on template and dev app. it must be sibling with <abp-loader-bar />)
- [ ] Work with design team ( @armgnunlu )

### PR DESCRIPTION
**Pull request by Sinan997**
Resolves #17008 

### Description

#17008 is already resolved but we are adding a signal alternative instead of using rxjs

### CODE REVIEW

File: npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-import { InternetConnectionService, LocalizationModule } from '@abp/ng.core';
+import { Component, inject  } from '@angular/core';
+import { NgIf, AsyncPipe } from '@angular/common'
+import { InternetConnectionSignalService , LocalizationModule } from '@abp/ng.core';
 
 @Component({
   selector: 'abp-internet-status',
   standalone: true,
-  imports:[CommonModule,LocalizationModule],
+  imports:[NgIf, AsyncPipe, LocalizationModule],
```
Comment (masum-ulu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts]: If we gonna use signal, don't need to import `AsyncPipe`

File: npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts [hunk 2]
```diff
@@ -24,11 +27,6 @@ export class InternetConnectionService{
     this.status.set(navigator.onLine)
     this.status$.next(navigator.onLine)
   }
-  
-  /* returns READONLY ANGULAR SIGNAL */
-  get networkStatus(){
-    return computed(() => this.status())
-  }
 
   /* returns OBSERVABLE */
```
Comment (masum-ulu) [inline npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts]: No need comment

File: npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts [hunk 3]
```diff
@@ -14,6 +14,9 @@ export class InternetConnectionService{
 
   /* creates writableSignal */
   private status = signal(navigator.onLine);
+
+  /* READONLY ANGULAR SIGNAL */
```
Comment (masum-ulu) [inline npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts]: No need comment

File: docs/en/UI/Angular/Internet-Connection-Service.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# Internet Connection Service
+`InternetConnectionService` is a service which is exposed by `@abp/ng.core` package. **If you want to detect whether you are connected to the internet or not, you can use this service.**
+
+
+### internet-connection-service.ts
+```
```
Comment (masum-ulu) [inline docs/en/UI/Angular/Internet-Connection-Service.md]: There is missing content type here. Please add **js** key to display much better content 

### Example

```js
import { Injectable } from  '@angular/core';

@Injectable({ providedIn: 'root' }) 
export  class  InternetConnectionService {}
```

File: docs/en/UI/Angular/Internet-Connection-Service.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 
 ### internet-connection-service.ts
```
Comment (masum-ulu) [inline docs/en/UI/Angular/Internet-Connection-Service.md]: There are a few bad UI please format example service 

image

**Comment by rqx110** [comment]
`Signal` is still a developer preview feature(angular 16).  and we have not migrate to `Signal`. 
do we need merge this to **7.4-preview** milestone?

https://github.com/abpframework/abp/issues/16670

**Comment by mahmut-gundogdu** [comment]
> `Signal` is still a developer preview feature(angular 16). and we have not migrate to `Signal`. do we need merge this to **7.4-preview** milestone?
> 
> #16670

Signal is developer preview yes. But it is simple component and we appened to source code, so if the user want to change , the user can do that.  Also the pr will released 4 weeks later, I think, Signal will marked as Stable.

for a dev = 7.4 preview now. We didn't create a 7.4 branch. I thought the PR target branch was right.

File: npm/ng-packs/packages/core/src/lib/tests/internet-connection.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    service = TestBed.inject(InternetConnectionService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('#networkStatus should return real value (true | false)', () => {
+    expect(typeof service.networkStatus()).toEqual('boolean')
+    expect(typeof !service.networkStatus()).toEqual('boolean')
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/core/src/lib/tests/internet-connection.service.spec.ts]: The test is pointless

File: npm/ng-packs/packages/core/src/lib/tests/internet-connection.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  it('#networkStatus$ should return real value (true | false)', () => {
+    service.networkStatus$.subscribe(val=>{
+      expect(typeof val).toEqual('boolean')
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/core/src/lib/tests/internet-connection.service.spec.ts]: you should control with test default value is correct and when you set a value, it should be equal to your value.

File: npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts [hunk 8]
```diff
@@ -42,7 +41,7 @@ import { InternetConnectionService, LocalizationModule } from '@abp/ng.core';
     }
   `]
 })
-export class  InternetConnectionStatusComponent{
-  internetConnectionService = inject(InternetConnectionService)
-  isOffline$ = this.internetConnectionService.networkStatus$.pipe(map((val) => !val));
+export class InternetConnectionStatusComponent{
+  internetConnectionService = inject(InternetConnectionService);
+  isOnline = this.internetConnectionService.networkStatus
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts:46]: you can use signal compute or converting opposite of value.  it just a note. it works like that.

**Review by mahmut-gundogdu — CHANGES_REQUESTED** [review]

**Review by mahmut-gundogdu — APPROVED** [review]

File: docs/en/UI/Angular/Internet-Connection-Service.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+1. Signal (readonly)
+2. Observable
+
+
+# How To Use
+İt's easy just inject the service and get network status
+
+**You can get via signal**
+```ts
+class someComponent{
```
Comment (masum-ulu) [inline docs/en/UI/Angular/Internet-Connection-Service.md]: class name must be [PascalCase](https://www.theserverside.com/definition/Pascal-case)

File: docs/en/UI/Angular/Internet-Connection-Service.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+ internetConnectionService  =  inject(InternetConnectionService);
+ isOnline = this.internetConnectionService.networkStatus
+}
+```
+**or you can get as observable**
+```ts
+class someComponent{
```
Comment (masum-ulu) [inline docs/en/UI/Angular/Internet-Connection-Service.md]: class name must be [PascalCase](https://www.theserverside.com/definition/Pascal-case)

File: docs/en/UI/Angular/Internet-Connection-Service.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
+# Internet Connection Service
+`InternetConnectionService` is a service which is exposed by `@abp/ng.core` package. **If you want to detect whether you are connected to the internet or not, you can use this service.**
+
+
+### internet-connection-service.ts
```
Comment (masum-ulu) [inline docs/en/UI/Angular/Internet-Connection-Service.md]: Remove this service section

**Review by braim23 — APPROVED** [review]

**Pull request merged**